.github/workflows/deploy.yml

```diff
@@ -1,43 +1,33 @@
-name: CI/CD Pipeline
+name: Deploy
 
 on:
   push:
     branches:
       - master
-    tags:
-      - "v*"
-  pull_request:
   workflow_dispatch:
+  workflow_run:
+    workflows: Test
+    types: completed
 
 jobs:
-  run-unit-tests:
+  get-version:
+    if: (github.ref == 'refs/heads/master' && github.event_name == 'push') || github.event_name == 'workflow_dispatch'
     runs-on: ubuntu-latest
-
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
 
-      - name: Setup .NET
-        uses: actions/setup-dotnet@v4
-        with:
-          dotnet-version: "8.x"
-
-      - name: Install dependencies
-        run: dotnet restore
-
-      - name: Build solution
-        run: dotnet build
-
-      - name: Run tests
+      - name: Get version
+        id: get-version
         run: |
-          dotnet test Tests/Tests.csproj
+          VERSION=$(cat version.txt)
+          echo "version=v$VERSION" >> "$GITHUB_OUTPUT"
+    outputs:
+      version: ${{ steps.get-version.outputs.version }}
 
   build-and-push-image:
-    needs: run-unit-tests
-    # If we push to master or manually run this workflow
-    if: (github.ref == 'refs/heads/master' && github.event_name == 'push') || github.event_name == 'workflow_dispatch'
+    needs: get-version
```

Comment:
Should we still have this if here?
`if: (github.ref == 'refs/heads/master' && github.event_name == 'push') || github.event_name == 'workflow_dispatch'`

Reply:
we have that if for the get-version job, which this build-and-push-image job relies on